server/src/language-service/diagnosticsProvider.ts

```diff
@@ -16,9 +16,21 @@ import {
     SepticReferenceProvider,
     AlgParsingError,
     AlgParsingErrorType,
+    SepticComment,
 } from "../septic";
 import { SettingsManager } from "../settings";
 
+export const disableDiagnosticRegex =
+    /\/\/\s+noqa\b(\s[\s\w,]+)?|\{#\s+noqa\b(\s[\s\w,]+)?\s*#\}/i;
+export const diagnosticCodeRegex = /E[0-9]{3}/;
+
+export enum DiagnosticCode {
```

Comment:
Another argument is that the settings for error/warning/hint/ignore aren't really that needed any longer since users can now disable individual diagnostics on a per-line level.

Reply:
Yeah, have been thinking that the settings options might not be necessary when you can turn it off using the noqa. Makes the code easier as well.